WFLY-21163 Upgrade Infinispan to 16.0.3 + WFLY-21164 Upgrade JGroups to 5.5 + WFLY-21266 Upgrade wildfly-clustering to 9.0

pferraro opened the pull request:
WFLY-21164 Upgrade JGroups to 5.5.2.Final.
WFLY-21266 Upgrade wildfly-clustering to 9.0.0.CR2

https://issues.redhat.com/browse/WFLY-21163
https://issues.redhat.com/browse/WFLY-21164
https://issues.redhat.com/browse/WFLY-21266

bstansberry commented:
@pferraro perhaps this is the cause of the CI failures?

ee-feature-pack/galleon-shared/src/main/resources/feature_groups/infinispan.xml:        <package name="org.infinispan.query.core"/>

pferraro committed:
WFLY-21163 Upgrade Infinispan to 16.0.3.
WFLY-21164 Upgrade JGroups to 5.5.2.Final.
WFLY-21266 Upgrade wildfly-clustering to 9.0.0.CR2.

pferraro commented:
> @pferraro perhaps this is the cause of the CI failures?
> ee-feature-pack/galleon-shared/src/main/resources/feature_groups/infinispan.xml:

Good catch.

rhusar commented on ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/infinispan/query/client/main/module.xml:
```diff
@@ -3,16 +3,17 @@
     </dependencies>
 
-</module>
+</module-->
```
Lets drop this.

pferraro committed:
Fix domain tests to accommodate changes to default protocol stacks.

pferraro committed:
Prune obsolete com.squareup.protoparser module.

rhusar commented:
> Is this something we think could be reviewable for WildFly 39 Beta or likely to still need some back and forth?

@darranl This is being actively reviewed, definitely for 39.x (adding missing label) and this should be in Beta for sure.

rhusar commented on testsuite/integration/clustering/pom.xml:
```diff
@@ -188,7 +183,7 @@
             <version>${legacy.version.com.squareup.okio}</version>
             <scope>test</scope>
-        </dependency>
+        </dependency-->
```
Optional: Great news that these are no longer needed, we can drop this.

pferraro committed:
Prune obsolete squareup test dependencies.

pferraro committed:
Prune obsolete reference to org.infinispan.query.core.

rhusar commented on clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerResourceDefinitionRegistrar.java:
```diff
@@ -269,7 +254,7 @@ public void registerSchema(SerializationContext context) {
                 // * This renders Infinispan default 2LC configuration unusable as it results in a default media type of application/unknown for keys and values
                 // See ISPN-12252 for details
-                builder.addModule(PrivateGlobalConfigurationBuilder.class).serverMode(!ServiceLoader.load(ModuleCommandExtensions.class, loader).iterator().hasNext());
+//                builder.addModule(PrivateGlobalConfigurationBuilder.class).serverMode(!ServiceLoader.load(ModuleCommandExtensions.class, loader).iterator().hasNext());
```
@pferraro Can you elaborate what's the situation with this? Isn't server mode already disabled by default, looking at `org.infinispan.configuration.internal.PrivateGlobalConfiguration#SERVER_MODE`? Can we just drop this block altogether?

pferraro committed:
Explicitly disable metrics, since we do not expose micrometer to the deployment.

pferraro committed:
Unconditionally enable server mode. Now that managed cache configuration always set key/value encoding, server mode is no longer incompatible with Hibernate 2LC interceptors.

darranl commented:
FYI I have submitted the following PR to combine the latest upstream changes with this PR so we can get it onto CI overnight.

https://github.com/wildfly/wildfly/pull/19477

Feel free to add to this PR if you would like to do it differently or if CI passes we can check if my PR can be approved.

rhusar reviewed (approved):
Review complete, discussed few things over the chat, LGTM!
